iterations_runners/iterations_runner.c

```diff
@@ -136,61 +156,40 @@ main(int argc, char **argv)
         }
 
         /* Start timed section */
-        krun_mperf_start = read_mperf();
-        krun_aperf_start = read_aperf();
-        krun_cycles_start = read_core_cycles();
-        krun_wallclock_start = clock_gettime_monotonic();
-
+        krun_measure(0);
         (void) (*krun_bench_func)(krun_param);
-
-        krun_wallclock_stop = clock_gettime_monotonic();
-        krun_cycles_stop = read_core_cycles();
-        krun_aperf_stop = read_aperf();
-        krun_mperf_stop = read_mperf();
+        krun_measure(1);
         /* End timed section */
 
-        /* Sanity checks */
-        if (krun_cycles_start > krun_cycles_stop) {
-            fprintf(stderr, "cycle count start greater than stop\n");
-            fprintf(stderr, "start=%" PRIu64 ", stop=%" PRIu64 "\n",
-                krun_cycles_start, krun_cycles_stop);
-            exit(EXIT_FAILURE);
-        }
+        /* Extract/check/store wallclock data from libkruntime */
+        krun_wallclock_start = krun_get_wallclock(0);
+        krun_wallclock_stop = krun_get_wallclock(1);
 
-        if (krun_wallclock_start > krun_wallclock_stop) {
-            fprintf(stderr, "wallclock time start greater than stop\n");
-            fprintf(stderr, "start=%f, stop=%f\n",
-                krun_wallclock_start, krun_wallclock_stop);
-            exit(EXIT_FAILURE);
-        }
+        krun_wallclock_times[krun_iter_num] =
+            krun_wallclock_stop - krun_wallclock_start;
 
-        if (krun_aperf_start > krun_aperf_stop) {
-            fprintf(stderr, "aperf start greater than stop\n");
-            fprintf(stderr, "start=%" PRIu64 ", stop=%" PRIu64 "\n",
-                krun_aperf_start, krun_aperf_stop);
-            exit(EXIT_FAILURE);
-        }
+        /* Same for per-core measurements */
+        for (krun_core = 0; krun_core < krun_num_cores; krun_core++ ) {
```

Comment:
Similar comment to earlier. I don't think the intermediate variables make things clearer (given that they're only used once).